patientIdentifierType.md

```diff
@@ -0,0 +1,104 @@
+
+* validator: full class name of an [IdentifierValidator](https://docs.openmrs.org/doc/org/openmrs/patient/IdentifierValidator.html) (e.g., [`org.openmrs.patient.impl.LuhnIdentifierValidator`](https://docs.openmrs.org/doc/org/openmrs/patient/impl/LuhnIdentifierValidator.html))
+
+* locationBehavior: "REQUIRED" if a location must be associated with the identifier; "NOT_USED" if the identifier does require a location
+
+## Available operations
+
+1. [List PatientIdentifierType resources](#List-patientidentifierType-resource)
+2. [Create a patientIdentifierType record](#create-a-patientidentifierType)
+3. [Update a patientIdentifierType record](#update-a-patientIdentifierType)
+4. [Delete a patientIdentifierType record](#delete-a-patientIdentifierType)
+
+### List PatientIdentifierType resource
+
+* #### List patientIdentifierType
+
+    Fetch all non-retired patientIdentifierTypes resources that match any specified parameters otherwise fetch all non-retired patients. Returns a `200 OK` status with the patientIdentifierType response. If the user is not logged in it returns `401 Unauthorized` status is returned.
+
+    ```console
+    GET /patientidentifiertype?
+        &q=**searchqQuery**
+    ```
+
+* #### List patientIdentifierType by UUID.
+
+    Retrieve a patientIdentifierType by its UUID. Returns a `404 Not Found` status if patientIdentifierType does not exist in the system. If the user is not logged in to perform this action, a `401 Unauthorized` status is returned.
+
+    ```console
+    GET /patientidentifiertype/:target_patientIdentifierType_uuid
+    ```
+### Create a patientIdentifierType
+
+* To create a patientIdentifierType you need to specify the below properties in the request. If you are not logged in to perform this action, a `401 Unauthorized` status is returned.
+
+    #### Properties
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *name* | string | label for the identifier
+    *description* | string | a small description about the patientIdentifier
+    *format* | string | a regular expression defining what the identifier text should contain
+    *formatDescription* | string | an optional description of the regular expression
+    *required* | boolean | a true/false whether every patient MUST have this type
+    *checkDigit* | boolean | a true/false whether this identifier has a checkdigit at the end
+    *validator* | string | `org.openmrs.patient.IdentifierValidator`
+    *locationBehavior* | "REQUIRED" or "NOT USED" | behavior of the location with respect to the identifier 
+    *uniquenessBehavior* | string | specify the uniqueness of the behaviour, it can be either Unique, Non Unique or Location.
+
+    ```console
+    POST /patientidentifiertype
+    {
+        "name": "Wilson Hospital Medical Record Number",
+        "description": "Wilson Hospital Medical Record Number is used as a patientIdentifier type.",
+        "format": "234-7",
```

Comment:
As your documentation explains, `format` is "a regular expression defining what the identifier text should contain." The [regular expression](https://www.regular-expressions.info/) "234-7" only matches a single string: the literal value "234-7". To make matters worse, not even the exact value would be accepted, since it has an invalid [check digit](https://om.rs/checkdigit).

I would suggest:
> "format": "\d{1,10}-\d",

The regular expression `\d{1,10}-\d` matches 1-10 digit(s) (`\d{1,10}`) followed by a literal hyphen (`-`) and a single digit. If you want to limit the medical record system to 1000 patients by using the pattern NNN-N, the corresponding regular expression would be `\d{3}-\d`.